app/grandchallenge/challenges/models.py

```diff
@@ -1402,3 +1403,81 @@ class ChallengeRequestGroupObjectPermission(GroupObjectPermissionBase):
     content_object = models.ForeignKey(
         ChallengeRequest, on_delete=models.CASCADE
     )
+
+
+class TaskResponsibleChoices(models.TextChoices):
+    SUPPORT = "SUP", "Support"
+    CHALLENGE_ORGANIZERS = "ORG", "Challenge Organizers"
+
+
+class OnboardingTask(FieldChangeMixin, UUIDModel):
+    ResponsibleChoices = TaskResponsibleChoices
+
+    class Meta:
+        permissions = [
+            ("complete_onboaringtask", "Can mark this task as completed")
+        ]
+
+    created = models.DateTimeField(editable=False)
+    challenge = models.ForeignKey(
+        to=Challenge,
+        on_delete=models.CASCADE,
+        related_name="onboarding_tasks",
+    )
+    title = models.CharField(
+        max_length=255,
+        help_text="Title of this task",
+    )
+
+    description = models.TextField(
+        blank=True, help_text="Description of this task."
+    )
+    complete = models.BooleanField(
+        default=False,
+        help_text="If true, this task is complete.",
+    )
+    completed_at = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text="Time this task was last marked completed.",
+    )
+    responsible = models.CharField(
+        blank=False,
+        max_length=3,
+        choices=ResponsibleChoices.choices,
+        default=ResponsibleChoices.CHALLENGE_ORGANIZERS,
+        help_text="Who is responsible for completion of this task.",
+    )
+    deadline = models.DateTimeField(
+        help_text="Deadline for this task.",
+    )
+
+    def is_overdue(self, reference_time):
+        return not self.complete and self.deadline < reference_time
```

Comment:
Wouldn't the `reference_time` always just be now if you want to know if a task is overdue at any given point in time?

Reply:
Practically: yes. Technically: no.

The `now()` should be a fixed reference or else you have slightly varying notion of overdue per Task. If two tasks are in the list, having the same deadline, but one is at the start of the list and the other at the end. The one at the start could be considered on time and the task at the end could be considered overdue.

Reply:
A bit of a theoretical cornercase, I think, since the difference in a list of say 25 tasks (our default pagination length in the UI) would be in the millisecond range, so unlikely to make a difference, right?   But I understand the theoretical point, so fine.